x/datadeal/keeper/deal.go

```diff
@@ -187,9 +187,9 @@ func (k Keeper) SellData(ctx sdk.Context, msg *types.MsgSellData) error {
 		return sdkerrors.Wrapf(types.ErrSellData, "deal status is not ACTIVE")
 	}
 
-	getDataSale, _ := k.GetDataSale(ctx, msg.DataHash, msg.DealId)
-	if getDataSale != nil && getDataSale.Status != types.DATA_SALE_STATUS_VERIFICATION_FAILED {
-		return sdkerrors.Wrapf(types.ErrSellData, "data already exists")
+	err = k.checkDataSaleStatus(ctx, msg.SellerAddress, msg.DataHash, msg.DealId)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrSellData, err.Error())
```

Comment:
This is only code style, I prefer below :)
```suggestion
	
	if err := k.checkDataSaleStatus(ctx, msg.SellerAddress, msg.DataHash, msg.DealId); err != nil {
		return sdkerrors.Wrapf(types.ErrSellData, err.Error())
```